lib/string/strcpy/strncpytail.h

```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: 2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_STRING_STRCPY_STRNCPYTAIL_H_
+#define SHADOW_INCLUDE_LIB_STRING_STRCPY_STRNCPYTAIL_H_
+
+
+#include "config.h"
+
+#include <stddef.h>
+#include <string.h>
+#include <sys/param.h>
+
+#include "attr.h"
+#include "sizeof.h"
+#include "string/strchr/strnul.h"
+
+
+#define STRNCPYTAIL(dst, src)  strncpytail(dst, src, countof(dst))
+
+
+ATTR_STRING(2)
+inline char *strncpytail(char *restrict dst, const char *restrict src,
+    size_t dsize);
+
+
+// nonstring copy tail-of-string
```

Comment:
@hallyn 

I'm standing by this term.  For the mnemonic comment explaining the name, nonstring is the most appropriate thing to use.  You can think of `strn` as "string-non".